src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -213,8 +225,64 @@ class ADRController(private val encryption: Encryption,
                 else -> adr.postFile("resource_patch", commonParameters + listOf("id" to resourceId), filePart)
             }
         }
-        finally {
+        finally
+        {
             tmpDir.deleteRecursively()
         }
     }
+
+    @Suppress("ReturnCount")
+    private fun makeOutputFileToPushToADR(artefact: Pair<ResponseEntity<StreamingResponseBody>, String>,
+                                          tmpDir: File,
+                                          resourceFileName: String): Pair<File?, ErrorDetail?>
+    {
+        // Return error if artefact can't be retrieved
+        if (!artefact.first.statusCode.is2xxSuccessful)
+        {
+            val baos = ByteArrayOutputStream()
+            artefact.first.body?.writeTo(baos)
+            return Pair(null, ErrorDetail(artefact.first.statusCode, baos.toString()))
+        }
+
+        // Stream artefact to file
+        val file = File(tmpDir, resourceFileName)
+        FileOutputStream(file).use { fis ->
+            artefact.first.body!!.writeTo(fis)
+        }
+        return Pair(file, null)
+    }
+
+    @Suppress("ReturnCount")
+    private fun makeInputFileToPushToADR(resourceType: String,
+                                         resourceId: String?,
+                                         tmpDir: File): Pair<File?, ErrorDetail?>
+    {
+        if (resourceId == null)
+        {
+            return Pair(null,
+                    ErrorDetail(HttpStatus.BAD_REQUEST, "resourceId must be provided for input resourceType"))
+        }
+
+        // Map adr schema to version file type
+        val fileType = when (resourceType)
+        {
+            appProperties.adrPJNZSchema -> FileType.PJNZ
```

Comment:
Slightly surprising that we haven't needed this mapping from schema to filetype elsewhere. If there's a problem here it's that we're using (untyped) properties values when they should really be an enum. But even if that's worth addressing then it's another ticket.

Reply:
I'm going to make this mapping into a single method called at the start of the post method, and use the FileType enum subsequently. I think we've avoided the mapping previous because most requests up til now are for single files of a known type.